pkg/cfn/manager/api.go

```diff
@@ -237,6 +239,92 @@ func (c *StackCollection) createStackRequest(stackName string, resourceSet build
 	return stack, nil
 }
 
+func (c *StackCollection) PropagateManagedNodeGroupTagsToASG(ngName string, ngTags map[string]string, asgNames []string, errCh chan error) error {
+	go func() {
+		defer close(errCh)
+		// build the input tags for all ASGs attached to the managed nodegroup
+		asgTags := []astypes.Tag{}
+
+		for _, asgName := range asgNames {
+			// skip directly if not tags are required to be created
+			if len(ngTags) == 0 {
+				continue
+			}
+
+			// check if the number of tags on the ASG would go over the defined limit
+			if err := c.checkASGTagsNumber(ngName, asgName, ngTags); err != nil {
+				errCh <- err
+				return
+			}
+			// build the list of tags to attach to the ASG
+			for ngTagKey, ngTagValue := range ngTags {
+				asgTag := astypes.Tag{
+					ResourceId:        aws.String(asgName),
+					ResourceType:      aws.String("auto-scaling-group"),
+					Key:               aws.String(ngTagKey),
+					Value:             aws.String(ngTagValue),
+					PropagateAtLaunch: aws.Bool(false),
+				}
+				asgTags = append(asgTags, asgTag)
+			}
+		}
+
+		// consider the maximum number of tags we can create at once...
+		var chunkedASGTags [][]astypes.Tag
+		chunkSize := builder.MaximumCreatedTagNumberPerCall
+		for start := 0; start < len(asgTags); start += chunkSize {
+			end := start + chunkSize
+			if end > len(asgTags) {
+				end = len(asgTags)
+			}
+			chunkedASGTags = append(chunkedASGTags, asgTags[start:end])
+		}
+		// ...then create all of them in a loop
+		for _, asgTags := range chunkedASGTags {
+			input := &autoscaling.CreateOrUpdateTagsInput{Tags: asgTags}
+			if _, err := c.asgAPI.CreateOrUpdateTags(context.Background(), input); err != nil {
+				errCh <- errors.Wrapf(err, "creating or updating asg tags for managed nodegroup %q", ngName)
+				return
+			}
+		}
+		errCh <- nil
+	}()
+	return nil
+}
+
+// checkASGTagsNumber limit considering the new propagated tags
+func (c *StackCollection) checkASGTagsNumber(ngName, asgName string, propagatedTags map[string]string) error {
+	tagsFilter := &autoscaling.DescribeTagsInput{
+		Filters: []astypes.Filter{
+			{
+				Name:   aws.String("auto-scaling-group"),
+				Values: []string{asgName},
+			},
+		},
+	}
+	output, err := c.asgAPI.DescribeTags(context.Background(), tagsFilter)
+	if err != nil {
+		return errors.Wrapf(err, "describing asg %q tags for managed nodegroup %q", asgName, ngName)
+	}
+	asgTags := output.Tags
+	// intersection of key tags to consider the number of tags going
+	// to be attached to the ASG
+	uniqueTagKeyCount := len(asgTags) + len(propagatedTags)
```

Comment:
If there are duplicates, we should error right away like here: https://github.com/weaveworks/eksctl/blob/main/pkg/cfn/builder/nodegroup.go#L302

Reply:
As the ASG is automatically created, it has some usual tags on it such as the cluster-name it runs on for example. This is normal and we don't want to throw an error about it.

Reply:
Ah gotcha. That's interesting. I didn't think of that.